### ISSUE TITLE
Rule S3966: FP when using StreamWriter 

### ISSUE THREAD

**Issue by costin-zaharia-sonarsource** [SonarSource/sonar-dotnet#3644]
### Description

`StreamWriter` is closing and not disposing the underling stream. See: https://github.com/microsoft/referencesource/blob/master/mscorlib/system/io/streamwriter.cs#L269

Rule description needs to be updated too since it's based on this FP.

Source: https://github.com/SonarSource/sonar-dotnet/issues/209#issuecomment-703929235

### Repro steps

```
using (Stream stream = new FileStream("file.txt", FileMode.OpenOrCreate))
{
    using (StreamWriter writer = new StreamWriter(stream))  // Noncompliant: 'stream' will be disposed twice
    {
        // Use the writer object...
    }
}
```

### Expected behavior

There should be no diagnostic raised for `stream` in this case since it is not disposed twice.

### Actual behavior

A FP is raised reporting that the `stream` was disposed twice.

### Known workarounds

None.

### Related information

* C#/VB.NET Plugins Version: https://github.com/SonarSource/sonar-dotnet/releases/tag/8.13.1.21947
